webhook/mutation/pod_mutator.go

```diff
@@ -204,264 +224,485 @@ func isWellKnownWorkload(ownerRef metav1.OwnerReference) bool {
 		},
 		SecurityContext: sc,
 		VolumeMounts: []corev1.VolumeMount{
-			{Name: "oneagent-bin", MountPath: "/mnt/bin"},
-			{Name: "oneagent-share", MountPath: "/mnt/share"},
-			{Name: "oneagent-config", MountPath: "/mnt/config"},
-			{Name: "data-ingest-enrichment", MountPath: "/var/lib/dynatrace/enrichment"},
+			{Name: injectionConfigVolumeName, MountPath: "/mnt/config"},
 		},
 		Resources: *dk.InitResources(),
 	}
+	return ic
+}
+
+func (m *podMutator) getDeploymentMetadata(dk dynatracev1beta1.DynaKube) *deploymentmetadata.DeploymentMetadata {
+	var deploymentMetadata *deploymentmetadata.DeploymentMetadata
+	if dk.CloudNativeFullstackMode() {
+		deploymentMetadata = deploymentmetadata.NewDeploymentMetadata(m.clusterID, deploymentmetadata.DeploymentTypeCloudNative)
+	} else {
+		deploymentMetadata = deploymentmetadata.NewDeploymentMetadata(m.clusterID, deploymentmetadata.DeploymentTypeApplicationMonitoring)
+	}
+	return deploymentMetadata
+}
+
+func (m *podMutator) getBasePodName(pod *corev1.Pod) string {
+	basePodName := pod.GenerateName
+	if basePodName == "" {
+		basePodName = pod.Name
+	}
+
+	// Only include up to the last dash character, exclusive.
+	if p := strings.LastIndex(basePodName, "-"); p != -1 {
+		basePodName = basePodName[:p]
+	}
+	return basePodName
+}
+
+func (m *podMutator) getSecurityContext(pod *corev1.Pod) *corev1.SecurityContext {
+	var sc *corev1.SecurityContext
+	if pod.Spec.Containers[0].SecurityContext != nil {
+		sc = pod.Spec.Containers[0].SecurityContext.DeepCopy()
+	}
+	return sc
+}
+
+func (m *podMutator) ensureDataIngestVolumes(injectionInfo *InjectionInfo, pod *corev1.Pod) {
+	if injectionInfo.enabled(DataIngest) {
+		pod.Spec.Volumes = append(pod.Spec.Volumes,
+			corev1.Volume{
+				Name: dataIngestVolumeName,
+				VolumeSource: corev1.VolumeSource{
+					EmptyDir: &corev1.EmptyDirVolumeSource{},
+				},
+			},
+			corev1.Volume{
+				Name: dataIngestEndpointVolumeName,
+				VolumeSource: corev1.VolumeSource{
+					Secret: &corev1.SecretVolumeSource{
+						SecretName: dtingestendpoint.SecretEndpointName,
+					},
+				},
+			},
+		)
+	}
+}
+
+func (m *podMutator) ensureOneAgentVolumes(injectionInfo *InjectionInfo, pod *corev1.Pod, dkVol corev1.VolumeSource) {
+	if injectionInfo.enabled(OneAgent) {
+		pod.Spec.Volumes = append(pod.Spec.Volumes,
+			corev1.Volume{Name: oneAgentBinVolumeName, VolumeSource: dkVol},
+			corev1.Volume{
+				Name: oneAgentShareVolumeName,
+				VolumeSource: corev1.VolumeSource{
+					EmptyDir: &corev1.EmptyDirVolumeSource{},
+				},
+			},
+		)
+	}
+}
 
+func (m *podMutator) ensureInjectionConfigVolume(pod *corev1.Pod) {
+	pod.Spec.Volumes = append(pod.Spec.Volumes,
+		corev1.Volume{
+			Name: injectionConfigVolumeName,
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: dtwebhook.SecretConfigName,
+				},
+			},
+		},
+	)
+}
+
+func (m *podMutator) getBasicData(pod *corev1.Pod) (string, string, string, string, string) {
```

Comment:
When you are returning 5 things you should either give the return values names 
`func (m *podMutator) getBasicData(pod *corev1.Pod) (technologies, installPath, installerURL, failurePolicy, image string)`
or create a struct (`OneAgentInstallConfig` or something) that has these fields